### ISSUE TITLE
Bridge settings, an idea for a new feature

### ISSUE THREAD

**Issue by dobrzann** [alexrj/Slic3r#3678]
this is a new feature request.

When the bridges are too long, lines are able to connect each other.
So maybe add a new function "Bridge settings"
-Bridge pattern angle
-Bridge pattern spacing (like in support settings)

is it possible?

**Comment by bubnikv** [comment]
Frankly I don't understand. Would you please draw a picture and give more
detail?

**Comment by dobrzann** [comment]
I mean, option to change spaces between bridge lines
![slic3r bridge](https://cloud.githubusercontent.com/assets/25279164/22185229/0516b074-e0e2-11e6-9808-adc7e8136ea5.jpg)

**Comment by landodragon141** [comment]
I'm pretty sure this does that.
<img width="405" alt="capture" src="https://cloud.githubusercontent.com/assets/1440561/22185716/ec1580bc-e0a7-11e6-89fc-493d002b50c0.PNG">

**Comment by dobrzann** [comment]
I think, you can change only extrusion width in this way (except extrusion width settings). Space between lines is the same. 
I think about something like "interface pattern spacing" in support options.

**Comment by lordofhyphens** [comment]
What the bridge fill ratio does is extrude less plastic for the same traces.

**Comment by jkoljo** [comment]
Yep, nowadays hot plastic from previous extrusion line can stick to nozzle, ruining the nice bridge. It would be great to be able to extrude every second line. And when those are done, do the rest, so lines have had time to cool down

**Comment by dobrzann** [comment]
is it a problem to add this feature?

**Comment by lordofhyphens** [comment]
@dobrzann it isn't necessarily a problem to add it in code, but we prefer to understand the problem  (and everything around it) before jumping to a solution.

Often what people ask for is not actually what they want. ;)

**Comment by forReason** [comment]
I believe what he wants is the folloing:
currently the nozzle will bridge one line of plastic and continue with the next line beneath it.

This causes the first line to stay heated and "melt down".

he wants to print the first line, then let out one or to to and bridge the next line. Later one add the skipped lines so the plastic had time to cool down.

not sure tho if this would work and/or be useful

**Comment by lordofhyphens** [comment]
I'd be worried about anchoring the bridge for those filler lines.

**Comment by dobrzann** [comment]
> Only print every Nth line of the bridge so the nozzle wont come near previously printed lines and ruin them.
    Continue with the next layer building onto the already hardened lines

And this is exactly what I mean. But It can be Nth line of the bridge or defined space between lines, like "interface pattern spacing" as I mentioned before

**Comment by Tinchus2009** [comment]
I understand what @dobrzann is asking for but, I say this with respect, @dobrzann wouldnt be better t have your bridging settings properly calibrated? With actual setting and proper calibrating (good bridge speed calibration, a good layer cooling and the correct bridging extrusion value) U can produce HUGE bridges (I have posted in thingiverse a bridging test of 15 cms long, with no sagging at all). 
So the problem I think is not a setting in slic3r but  correct tunning for bridging.

**Comment by kefir-** [comment]
I agree that proper calibration is important, but I think different nozzles can cause different issues. I suspect smaller nozzle sizes can cause more trouble for bridging, and different nozzle designs may also have some impact on how the molten flow leaves the nozzle. And it'd be interesting to have some way to experiment further when no changes of speed, temperature and flow rate have managed to stop an otherwise well calibrated printer's nozzle from mopping up the previous bridge strand, for example.

**Comment by lordofhyphens** [comment]
Thing is that the bridging flow is supposed to leave a good surface to
print on.

If you widen the space between the traces, you at least need another bridge
flow in the opposite direction that has the tight spacing in order to print
on it.

I can probably throw together a short hack that will do something like this
for people to experiment with, but I personally don't believe that it is
going to be a worthwhile endeavor.

**Comment by Tinchus2009** [comment]
i agree about noozle design and its effects on printing but then we will be
talking about how to solve through software a hardware fedign problem. and
to me, bad dezigned/cheap hardwared has its prize. on the other hand, the
idea being proposed as a solution willbringnew problems: if 1 bridge fails
there will be a very nice hole in the surface. also, for haveing a decent
surface finish afyer a bridged area you need at least 3 layers.  with this
idea that number will for sure be increased.

**Comment by dobrzann** [comment]
> I can probably throw together a short hack that will do something like this for people to experiment with, but I personally don't believe that it is going to be a worthwhile endeavor.

Please do it if you can.
Maybe it worth a try.

**Comment by bubnikv** [comment]
I consider the idea to be sound. One cannot say whether it will work until one tries it.

I actually had the same idea some months ago and it is somewhere at the bottom of my veeery long list.

> if 1 bridge fails there will be a very nice hole in the surface. also, for haveing a decent surface finish afyer a bridged area you need at least 3 layers.  with this idea that number will for sure be increased.

The idea is to put down every 2nd layer and then to press the intermediate threads at the same layer in between the already placed threads. Maybe the 1st set of threads could be printed with the bridging flow with a bit less than twice the spacing, and the 2nd set of threads with a bit less material and pressed in between the already laid threads. My hopes are, that one would actually get a better filled bridge surfaces.

**Comment by dobrzann** [comment]
I see the topic died :-)

@lordofhyphens 
```
I can probably throw together a short hack that will do something like this
for people to experiment with, but I personally don't believe that it is
going to be a worthwhile endeavor.

```

Is there any chance you will add this function for the tests?

**Comment by sachel85** [comment]
Anyone still working on this?

**Comment by lordofhyphens** [comment]
I am not working on it. I am too busy.

@sachel85 if you want to take a run at it be my guest.
